api/v1beta1/proposal_webhook.go

```diff
@@ -169,8 +165,13 @@ func validateProposalSource(ctx context.Context, c client.Client, proposalSource
 			return err
 		}
 	}
+	if proposalSource.UpgradeChaincode != nil {
+		if err := validateChaincodePhase(ctx, c, proposalSource.UpgradeChaincode.Chaincode); err != nil {
+			return err
+		}
+	}
 
-	return nil
+	return validateChaincodeBuild(ctx, c, proposalSource)
```

Comment:
nit: For readability, it is recommended that we should put this function in `if` like others.